src/index.cpp

```diff
@@ -2892,6 +2942,29 @@ template <typename T, typename TagT, typename LabelT> void Index<T, TagT, LabelT
 
     reposition_points((uint32_t)_nd, (uint32_t)_max_points, (uint32_t)_num_frozen_pts);
     _start = (uint32_t)_max_points;
+
+    if (_filtered_index && _dynamic_index)
+    {
+        reposition_medoids((uint32_t)_nd, (uint32_t)_max_points);
+    }
+}
+
+template <typename T, typename TagT, typename LabelT>
```

Comment:
comment on what this function does

Reply:
Added the comment.
/* updates medoid id's in case of filtered dynamic index as frozen_points acts as medoids.
 * is called after reposition_frozen_points().
 */
template <typename T, typename TagT, typename LabelT>
void Index<T, TagT, LabelT>::reposition_medoids(uint32_t old_location_start, uint32_t new_location_start){......}